crates/trans_token/src/event.rs

```diff
@@ -0,0 +1,326 @@
+//! Token transaction events.
+
+use std::borrow::Cow;
+use std::fmt;
+use std::str::FromStr;
+
+use namada_core::address::Address;
+use namada_core::uint::Uint;
+use namada_events::extend::{Closure, ComposeEvent, EventAttributeEntry};
+use namada_events::{Event, EventLevel, EventToEmit, EventType};
+
+pub mod types {
+    //! Token event types.
+
+    use namada_events::{event_type, EventType};
+
+    use super::TokenEvent;
+
+    /// Mint token event.
+    pub const MINT: EventType = event_type!(TokenEvent, "mint");
+
+    /// Burn token event.
+    pub const BURN: EventType = event_type!(TokenEvent, "burn");
+
+    /// Transfer token event.
+    pub const TRANSFER: EventType = event_type!(TokenEvent, "transfer");
+}
+
+/// A user account.
+#[derive(Debug, Clone, Ord, PartialOrd, Eq, PartialEq, Hash)]
+pub enum UserAccount {
+    /// Internal chain address in Namada.
+    Internal(Address),
```

Comment:
Can this cause some confusion with the enum variant we have on the `Address` type?

Reply:
hmm.. any other name suggestions?

Reply:
I'm terrible at naming. Let's keep it like that 👍🏻